### PR TITLE
Add Linux "Wire" menu, add fullscreen and menu bar toggles

### PR DESCRIPTION
**Pull request by ConorIA**
This is very, very basic, but I thought I'd take a shot at #71. All of the 4 items work.

menu

Note that I thought about adding the fullscreen options to the Linux menu as you've done for Mac, but I couldn't figure out how to get back to windowed mode without editing my config files.

### CODE REVIEW

**Comment by ConorIA** [comment]
Just noticed something that isn't quite right: The quit menu item quits Wire, but the Ctrl-Q key combo just closes the window to the tray.

**Comment by lipis** [comment]
on mobile now.. but on Linux is more common to have alt+F4 for quit.. no?

**Comment by ConorIA** [comment]
Most apps use Ctrl-Q as the shortcut (I had a bunch of windows open anyway, so I tried it and it closed Nautilus, Atom, Thunderbird, Gedit, at least). Although Alt-F4 worked as well on Nautilus. 

Atom and Brave browser also use Ctrl-Q as the quit shortcut in menus. 

One thing that is certainly out of place in my example though is the preferences menu. Preferences are usually under "Edit" in Linux.

**Comment by lipis** [comment]
Right.. I checked it as well.. `Ctrl+Q` should be OK. Preferences should go under edit though for consistency..

File: electron/js/menu/system.js [hunk 1]
```diff
@@ -298,6 +315,10 @@ if (process.platform === 'win32') {
   });
 }
 
+if (process.platform === 'linux') {
+  menuTemplate.unshift(linuxTemplate);
```
Comment (lipis) [inline electron/js/menu/system.js]:
```js
  editTemplate.submenu.push(separatorTemplate, {
    i18n: 'menuPreferences',
    click: function() {sendAction('profile-settings-show');},
  });
```

Comment (lipis) [inline electron/js/menu/system.js]: something like that maybe..

**Comment by ConorIA** [comment]
I'll make those changes tomorrow. Tonight Snarky Puppy is playing in Toronto! 😋

**Comment by ConorIA** [comment]
There we go!

File: electron/js/menu/system.js [hunk 2]
```diff
@@ -267,6 +267,18 @@ var win32Template = {
   ],
 };
 
+var linuxTemplate = {
+  label: config.NAME,
```
Comment (ConorIA) [inline electron/js/menu/system.js]: I guess so, though it isn't really a 'traditional' menu, is it? 
To change that, would line 271 change to `i18n: 'menuFile',`?

Comment (ConorIA) [inline electron/js/menu/system.js]: Oh no! I just pushed changes to make the menu 'File' (though you might need to double-check my German!). Anyway, on Monday let me know what the designers prefer and I can revert if necessary.

**Comment by ConorIA** [comment]
Another quick observation from looking at other apps: Edit is virtually always beside File. 

That would mean that the Menus would be: 
Wire (or File), Edit, Conversation, Window, Help

I see where the menu order is specified in `system.js`, but I don't want to make any changes because it seems like that would change it for all apps. So I'll wait to hear from you whether that is a good thing to change or not. 

I'm not sure if it is just a coincidence, but most apps seem to use a hierarchy of: 

1) File, Edit, View
2) Application-specific menu items
3) Window, Help

File: electron/js/menu/system.js (hunk 2 above)
Comment (ConorIA) [inline electron/js/menu/system.js]: Sounds good. I'm happy to tinker. I am learning a lot!

**Comment by ConorIA** [comment]
In the end I decided to go ahead and push the changes from my last comment. It won't take much to revert them next week if necessary.

File: electron/js/menu/system.js [hunk 3]
```diff
@@ -267,9 +267,21 @@ var win32Template = {
...
+      accelerator: 'Ctrl+Q',
+      click: function() {app.quit();},
+    },
+  ],
+};
+
 menuTemplate = [
-  conversationTemplate,
   editTemplate,
+  conversationTemplate,
```
Comment (lipis) [inline electron/js/menu/system.js]: rollback the order.. if anything could happen later in another review/pr

File: electron/js/menu/system.js [hunk 4]
```diff
@@ -267,9 +267,21 @@ var win32Template = {
   ],
 };
 
+var linuxTemplate = {
+  label: config.NAME,
+  submenu: [
+    localeTemplate,
```
Comment (lipis) [inline electron/js/menu/system.js]: add a separator after locale

File: electron/js/menu/system.js [hunk 5]
```diff
@@ -271,6 +271,7 @@ var linuxTemplate = {
   label: config.NAME,
   submenu: [
     localeTemplate,
```
Comment (lipis) [inline electron/js/menu/system.js]: So before the languages add Preferences and separator.. here as well (without removing the preferences from Edit)

File: electron/js/menu/system.js [hunk 6]
```diff
@@ -298,6 +311,21 @@ if (process.platform === 'win32') {
...
 
+if (process.platform === 'linux') {
+  menuTemplate.unshift(linuxTemplate);
+  editTemplate.submenu.push(separatorTemplate, {
+    i18n: 'menuPreferences',
+    click: function() {sendAction('profile-settings-show');},
+  });
+  windowTemplate.submenu.push(
+    separatorTemplate,
+    toggleMenuTemplate,
```
Comment (lipis) [inline electron/js/menu/system.js]: remove the menu toggle from here.. doesn't seem to do much, there will be changes in Windows soon here so we'll rethink about it in the future

**Comment by lipis** [comment]
Still here?

**Comment by ConorIA** [comment]
Away from the computer! Won't be home until the evening. I think the option is enabled that should let you edit the PR, does that just mean edit the comments, or can you edit the commits?

**Comment by lipis** [comment]
or we can take it from here... thank you! :)

**Comment by ConorIA** [comment]
That works too! 😁

**Comment by lipis** [comment]
You are right.. didn't think of that :) Next time.. thanks

**Pull request merged**